bot/exts/info/information.py

```diff
@@ -547,7 +547,21 @@ async def rules(self, ctx: Context, rules: Greedy[int]) -> None:
         for rule in rules:
             self.bot.stats.incr(f"rule_uses.{rule}")
 
-        final_rules = tuple(f"**{pick}.** {full_rules[pick - 1]}" for pick in rules)
+        if rules:
+            final_rules = tuple(f"**{pick}.** {full_rules[pick - 1][0]}" for pick in rules)
+        else:
+            final_rules = tuple(f"**{pick + 1}** {full_rules[pick][0]}" for pick, rule in enumerate(full_rules)
+                                if keyword in rule[1])
```

Comment:
If both integers and keywords are supplied, I think it should search for both. What you wrote ignores keywords if an integer is supplied.

Reply:
@MarkKoz  Yes, true. And that was done on purpose after the discussion that @wookie184, others and I have had.
You can read that in issue #2108. 
The purpose is to be able to fetch rules on keywords as a secondary option, but if someone wants to use rule numbers, then it gets processed as a priority. Otherwise it can either conflict or confuse the person to use both.

Reply:
I'm not strongly against it as is. I don't think using both is inherently confusing, but it is likely a niche use case. It may become confusing only because we're trying to allow arbitrary text after the command and it may pick up a word as a keyword that was not intended to be a keyword.

Reply:
Done.